sdk/lib/opentelemetry/sdk/resources/auto_detector.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/sdk/resources/detectors/google_cloud_platform'
+require 'opentelemetry/sdk/resources/detectors/telemetry'
+
+module OpenTelemetry
+  module SDK
+    module Resources
+      class AutoDetector
```

Comment:
This doesn’t feel class-worthy. I think it could just be:
```ruby
module Resources
  def self.auto_detect
    ...
  end
end
```

Reply:
Agreed.